db/sql/05_msar.sql

```diff
@@ -698,6 +698,55 @@ END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
+CREATE OR REPLACE FUNCTION
+msar.get_schema_objects_table(schemas regnamespace[])
+RETURNS TABLE (obj_id oid, obj_schema text, obj_name text, obj_kind text) AS $$ /*
+Return a table with information about most objects in the given schemas.
+*/
+WITH obj_cte AS (
+  (
+    SELECT
+      oid AS obj_id,
+      msar.get_schema_name(pronamespace) AS obj_schema,
+      proname AS obj_name,
+      CASE prokind
+        WHEN 'a' THEN 'AGGREGATE'
+        WHEN 'p' THEN 'PROCEDURE'
+        ELSE 'FUNCTION'
+      END AS obj_kind
+    FROM pg_proc
+    WHERE pronamespace=ANY(schemas)
+  ) UNION (
+    SELECT
+      oid AS obj_id,
+      msar.get_schema_name(typnamespace) AS obj_schema,
+      typname AS obj_name,
+      'TYPE' AS obj_kind
+    FROM pg_type
+    WHERE typnamespace=ANY(schemas)
+  ) UNION (
+    SELECT
+      oid AS obj_id,
+      msar.get_schema_name(relnamespace) AS obj_schema,
+      relname AS obj_name,
+      CASE relkind
+        WHEN 'r' THEN 'TABLE'
+        WHEN 'p' THEN 'TABLE'
+        WHEN 'i' THEN 'INDEX'
+        WHEN 'I' THEN 'INDEX'
+        WHEN 'S' THEN 'SEQUENCE'
+        WHEN 'v' THEN 'VIEW'
+        WHEN 'm' THEN 'MATERIALIZED VIEW'
+        WHEN 'c' THEN 'TYPE'
+        WHEN 'f' THEN 'FOREIGN TABLE'
+      END AS obj_kind
+    FROM pg_class
+    WHERE relnamespace=ANY(schemas)
+  )
+) SELECT DISTINCT obj_id, obj_schema, obj_name, obj_kind FROM obj_cte WHERE obj_kind IS NOT NULL;
+$$ LANGUAGE SQL;
```

Comment:
Shouldn't this be `STABLE` & return null on null input?

```suggestion
$$ LANGUAGE SQL STABLE RETURNS NULL ON NULL INPUT;
```

Reply:
Sure, that works.